beancount.el

```diff
@@ -876,14 +878,42 @@ what that column is and returns it (an integer)."
   "Start a new timestamped directive with date DAYS before today."
   (interactive "P")
   (unless (bolp) (newline))
-  (insert (beancount--shift-current-date days) " "))
-
-(defun beancount--shift-current-date (days)
-  "Return ISO-8601 formatted date DAYS before today."
-  (let ((days-to-shift (- (or days 0))))
-    (format-time-string
-     "%Y-%m-%d"
-     (time-add (current-time) (days-to-time days-to-shift)))))
+  (insert (beancount--shift-date (current-time) (- (or days 0))) " "))
+
+(defun beancount--shift-date (today days)
+  "Return ISO-8601 formatted date TODAY shifted by DAYS.
+DAYS can negative or positive."
+  (format-time-string
+   "%Y-%m-%d"
+   (time-add today (days-to-time days))))
```

Comment:
It's kind of weird to have a function accept Emacs-epoch days but return a formatted string. I would prefer this function to take a formatted date or nil and return a formatted date. As the date parsing functions (surprisingly) clobber the match data, it should also use `(save-match-data ...)` around `date-to-time`. Bonus points for replacing `date-to-time` with some function more accurate for the task from `iso8601.el`.